### PR TITLE
Run 5373 nwi allow non hwnd uuids

### PR DESCRIPTION
**Pull request by dhamberlin**
#### Description of Change
Jira: https://appoji.jira.com/browse/RUN-5373

Checks `ProcessTracker`'s map of windows when performing `ExternalWindow` operations, to allow such operations on windows with a non-hwnd uuid.

#### Checklist
- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] PR title starts with the JIRA ticket [pull request process](https://github.com/openfin/Internal-Wiki/wiki/Pull-Request-Process)

### CODE REVIEW

**Comment by dhamberlin** [comment]
One thought on this flow:
Since we must pass an hwnd into ExternalWindow, perhaps it would make sense for `getNativeWindowInfoLite` to return an hwnd in addition to a uuid, to make it clearer which is which (even if they are the same in many cases).

File: src/browser/api/external_window.ts [hunk 1]
```diff
@@ -214,11 +215,16 @@ export function findExternalWindow(identity: Identity): Shapes.ExternalWindow |
 }
 
 /*
-  Checks whether an hwnd is a valid target for wrapping
+  Checks whether a uuid is a valid target for wrapping
 */
-function doesExternalWindowExist(uuid: string): boolean {
+function findExsistentNativeWindow(uuid: string): Shapes.NativeWindowInfoLite|undefined {
```
Comment (rdepena) [inline src/browser/api/external_window.ts]: guess this comment is !:neckbeard: 
`findExsistingNativeWindow` ? 

not a request for change

Comment (dhamberlin) [reply src/browser/api/external_window.ts]: ¯\_(ツ)_/¯ Either seems fine to me. Happy to conform to the will of the people.

Comment (tgoc99) [reply src/browser/api/external_window.ts]: No first s in either of those words... "Existing"

Comment (dhamberlin) [reply src/browser/api/external_window.ts]: Sorry guys, I'm bad at spelling apparently -.-
Thanks for looking out. Fixed.

File: src/browser/api/external_window.ts [hunk 2]
```diff
@@ -244,11 +245,16 @@ export function getRegisteredExternalWindow(identity: Identity): Shapes.External
...
+  Checks whether a uuid is a valid target for wrapping
 */
-function doesExternalWindowExist(uuid: string): boolean {
+function findExistingNativeWindow(uuid: string): Shapes.NativeWindowInfoLite|undefined {
   const allNativeWindows = electronApp.getAllNativeWindowInfo(true);
-  return !!allNativeWindows.find(win => getNativeWindowInfoLite(win).uuid === uuid);
+  // Check whether we are using a uuid generated by `launchExternalProcess`
+  const prelaunchedProcess = ProcessTracker.getProcessByUuid(uuid);
+  const win = prelaunchedProcess
+    ? allNativeWindows.find(win => win.process.pid === prelaunchedProcess.process.id)
```
Comment (MichaelMCoates) [inline src/browser/api/external_window.ts]: Small naming thing, but you're using just `win` here a lot. Maybe specify what each `win` is supposed to be?

**Comment by MichaelMCoates** [comment]
Do you have tests for this?

**Pull request merged**